src/common/Throttle.h

```diff
@@ -330,4 +331,183 @@ class OrderedThrottle {
   uint32_t waiters = 0;
 };
 
+
+template <typename T, typename I>
```

Comment:
Not necessary to template the entire class -- see ```librbd::ObjectMap``` for an example of how to create a templated callback handler

Reply:
okey done.